### PR TITLE
Adding support for auth for the registry client

### PR DESCRIPTION
**Pull request by akoserwal**
Authentication/authorization proof of concept: https://github.com/Apicurio/apicurio-registry/issues/674
Poc with keycloak:
https://github.com/carlesarnal/apicurio-registry/tree/auth-poc-674

Update Registry Client to support a client method with a configurable authentication mechanism. Config is provided as a parameter in the create/cached method. Currently, it only supports keycloak auth strategy, but it can be extended to support other mechanisms. 

Example:
```
String registryUrl = "http://0.0.0.0:8080/api";
AuthConfig config = new AuthConfig.Builder(AuthProvider.KEYCLOAK)
				.setServerUrl("http://127.0.0.1:8081/auth")
				.setClientId("admin-cli")
				.setRealm("master")
				.setUsername("admin")
				.setPassword("admin")
				.Build();

RegistryService service = RegistryClient.cached(registryUrl, config);

```

### CODE REVIEW

File: client/src/main/java/io/apicurio/registry/client/RegistryClient.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
  */
 public class RegistryClient {
     private static final Logger log = Logger.getLogger(RegistryClient.class.getName());
+    private static String token;
```
Comment (carlesarnal) [inline client/src/main/java/io/apicurio/registry/client/RegistryClient.java:34]: We should manage the auth logic for the client. It should take care of authentication and refreshing implementation so the users should not worry about auth stuff.

Comment (akoserwal) [reply client/src/main/java/io/apicurio/registry/client/RegistryClient.java]: @carlesarnal: Made changes, please review again

**Comment by carlesarnal** [comment]
Hi @akoserwal, thanks for the contribution! As we have commented, please, add some tests in the registry area regarding authentication so we can ensure that the client auth works properly.

File: client/pom.xml [hunk 2]
```diff
@@ -26,6 +26,31 @@
             <artifactId>resteasy-client-microprofile</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-admin-client</artifactId>
+            <version>10.0.1</version>
+            <exclusions>
+                <exclusion>
```
Comment (carlesarnal) [inline client/pom.xml:34]: Just curious, what's the reason for these exclusions?

Comment (akoserwal) [reply client/pom.xml:34]: Keycloak admin client uses version `3.9.1.Final` for rest-easy client. Which is conflicting with the versions already included by micro profile client (4.X) & other dependencies.

**Comment by carlesarnal** [comment]
Hi @akoserwal, any updates on this? Since we're going to replace the client, my idea is to reuse this to provide auth for the new one so you can contribute there too.

**Comment by akoserwal** [comment]
@carlesarnal: I am currently evaluating libraries & ways to test the auth flow by generating a dummy token and verify the token.

**Comment by carlesarnal** [comment]
Hey @akoserwal, what is the status of this? Do you think that we can merge it now?

**Comment by akoserwal** [comment]
@carlesarnal: Yes, we can merge it.

**Pull request merged**